src/aggregator/aggregator/aggregator.go

```diff
@@ -757,15 +703,8 @@ func (agg *aggregator) closeShardsAsync(shards []*aggregatorShard) {
 }
 
 func (agg *aggregator) tick() {
-	defer agg.wg.Done()
-
 	for {
-		select {
-		case <-agg.doneCh:
-			return
-		default:
-			agg.tickInternal()
-		}
+		agg.tickInternal()
```

Comment:
I guess this is just fine to let die when the process dies? Instead of waiting on a done signal?

Reply:
Yeah, Tick itself is pretty straightforward, it updates some metrics on how many series the aggregator currently has of each type, and expires old metrics from the local metric lists, which isn't necessary when the aggregator is being closed anyway

The metrics of an aggregator going down might look a little funky after this change, but that's to be expected anyway so not a huge deal

Reply:
I'll add a comment to this effect

Reply:
I'm cool with taking out the doneCh management in tickInternal() cause that is just too low level with little value, but the doneCh here gives this tick routine a chance to end, so you know all threads will be closed after aggregator is closed. It also allows you to comfortably do leaktest in unit tests, which feels like good hygiene to me.

Reply:
I think the idea is that when aggregator is closed, realistically the whole process is being killed in the next few seconds... it does end up being a fairly dirty close in terms of running threads etc, but the practical benefit of closing these in cleanup seems like a little much, overhead wise. If you feel strongly about this I can add doneCh cancellation here and avoid waiting on a wait group in the Close method, but don't think it will make much of a difference in regular operation

Reply:
Sounds good 👍 feel free to leave it out